triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala

```diff
@@ -116,30 +188,35 @@ object Server {
           // should be running as.  Returns a UUID for the newly
           // started trigger.
           path("v1" / "start") {
-            entity(as[StartParams]) {
-              params =>
-                Trigger.fromIdentifier(compiledPackages, params.identifier) match {
-                  case Left(err) =>
-                    complete(errorResponse(StatusCodes.UnprocessableEntity, err))
-                  case Right(trigger) =>
-                    val party = params.party
-                    val uuid = UUID.randomUUID
-                    val ident = uuid.toString
-                    val ref = ctx.spawn(
-                      TriggerRunner(
-                        new TriggerRunner.Config(
-                          compiledPackages,
-                          trigger,
-                          ledgerConfig,
-                          maxInboundMessageSize,
-                          party),
-                        ident),
-                      ident + "-monitor")
-                    triggers = triggers + (uuid -> TriggerRunnerWithParty(ref, party))
-                    val newTriggerSet = triggersByParty.getOrElse(party, Set()) + uuid
-                    triggersByParty = triggersByParty + (party -> newTriggerSet)
-                    val triggerIdResult = JsObject(("triggerId", uuid.toString.toJson))
-                    complete(successResponse(triggerIdResult))
+            extractRequest {
+              request =>
+                entity(as[StartParams]) {
+                  params =>
+                    Trigger.fromIdentifier(compiledPackages, params.identifier) match {
```

Comment:
The order here seems odd, I would expect that you first check the token and then the trigger parameter instead of the other way around.

Reply:
Got you. I'll reorder it.